docs/source/cuda/cudapysupported.rst

```diff
@@ -0,0 +1,202 @@
+========================================
+Supported Python features in CUDA Python
+========================================
+
+This page lists the Python features supported in the CUDA Python.  This includes
+all kernel and device functions compiled with ``@cuda.jit`` and other higher
+level Numba decorators that targets the CUDA GPU.
+
+Language
+========
+
+Execution Model
+---------------
+
+CUDA Python maps directly to the *single-instruction multiple-thread*
+execution (SIMT) model of CUDA.  Each instruction is implicitly
+executed by multiple threads in parallel.  With this execution model, array
+expression is less useful because we don't want multiple threads to perform
+the same task.  Instead, we want threads to perform a task in a cooperative
+fashion.
+
+The CUDA execution model also has some limitations.
+For instance, dynamic memory allocation is limited due to the
+requirement of knowing and configuring the maximum size of the heap.  The
+allocated memory is located in the slower global memory.  To minimize the
+overhead, CUDA Python avoids features that requires dynamic allocation.
+
+Another important semantic different is that there is
+no *global interpreter lock* (GIL) in CUDA Python.  Race conditions are possible
+when operating on writable shared data, which can be arrays in global memory
+or shared memory.
+
+There are several less apparent limitation on code complexity and
+size.  The high compute throughput of CUDA depends on threads operating on
+the same instruction due to SIMT.  When threads are diverged, hardware
```

Comment:
Whilst this is true, it's only true within a warp - could this be added to the description?